skrifa/src/meta/metrics.rs

```diff
@@ -0,0 +1,434 @@
+//! Global font and glyph specific metrics.
+
+use read_fonts::{
+    tables::{glyf::Glyf, hmtx::LongMetric, hvar::Hvar, loca::Loca},
+    types::{BigEndian, GlyphId},
+    TableProvider,
+};
+
+use crate::NormalizedCoord;
+
+/// Metrics for a text decoration.
+#[derive(Copy, Clone, PartialEq, Default, Debug)]
+pub struct Decoration {
+    /// Offset to the top of the decoration from the baseline.
+    pub offset: f32,
+    /// Thickness of the decoration.
+    pub thickness: f32,
+}
+
+/// Extents of a rectangular region.
```

Comment:
Do we not have the concept of a box in font-types? If not ... should we?

Reply:
Good call. I’ll move this there and make it generic over coordinate type.